tunder/lib/src/database/query.dart

```diff
@@ -16,9 +21,9 @@ class Query<T> {
 
   late bool _shouldMap = false;
   String? _orderBy;
-  List<Where> _wheres = [];
+  List<Where> wheres = [];
 
-  DatabaseConnection get _connection => DB.connection;
+  QueryOperation get _operation => QueryOperation.forDatabase(DB.driver);
```

Comment:
The switch statement inside the `forDatabase` factory constructor is going to work fine. However, a more object-oriented approach may be explored later on. Maybe by creating a `PostgresQuery` containing a `PostgresQueryOperation` for the `_operation`, `PostgresCountOperation` for the `count()` method and so on... The sole purpose of this change would be to integrate other database management systems more easily.

Reply:
That's a great suggestion. Feel free to create a discussion about this with a proposed API if you want to. But I definitely agree. 👍🏻 🚀